File: Java_Generated/Java/untitled/src/measurements/DataConsumerVariableMeasurementsStarted.java
package measurements;

import diagram.interfaces.IDesktop;
import general_service.AliasInitialValueCollection;
import general_service.interfaces.IInitialValueCollection;
import measurements.interfaces.IStarted;

public class DataConsumerVariableMeasurementsStarted extends DataConsumerVariableMeasurements implements IStarted {
    public DataConsumerVariableMeasurementsStarted(String name, IDesktop desktop) {
        super(name, desktop);
    }

    @Override
    public void startedStart(double startTime) {
        initial.resetInitialValues();
    }

    protected void createInitial()
    {
        initial = new AliasInitialValueCollection(this, this);
    }

    protected void setInitial()
    {
        initial = new AliasInitialValueCollection(this, this);
    }

    protected IInitialValueCollection initial;
}

File: src/TypeScript/TypeScriptLibrary/Library/Measurements/DataConsumerVariableMeasurementsStarted.ts
/* eslint-disable no-var */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { AliasInitialValueCollection } from "../AliasInitialValueCollection.";
import type { IAlias } from "../Interfaces/IAlias";
import type { IDesktop } from "../Interfaces/IDesktop";
import type { IInitialValueCollection } from "../Interfaces/IInitialValueCollection";
import { DataConsumerVariableMeasurements } from "./DataConsumerVariableMeasurements";
import type { IStarted } from "./Interfaces/IStarted";

export class DataConsumerVariableMeasurementsStarted extends DataConsumerVariableMeasurements implements IStarted
{
    protected initial !: IInitialValueCollection;


    protected alias !: IAlias;

    constructor(desktop: IDesktop, name: string)
    {
        super(desktop, name);
        this.typeName = "DataConsumerVariadbleMeasurementsStarted";
        this.types.push("IStarted");
        this.types.push("DataConsumerVariadbleMeasurementsStarted");
        this.alias = this;
    }


    startedStart(start: number): void {
        this.initial.resetInitialValues();
    }



    setInitial(): void {
        this.initial = new AliasInitialValueCollection(this, this);
    }

}